### PR TITLE
Re-factor DeviceProperties for better logging, exception handling and…

### PR DESCRIPTION
**Pull request by isaac73**
… comments

# Type of change? 

- [ ] Bug fix
- [ ] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 
Re-factor DeviceProperties for better logging, exception handling and comments
...

**Checklist:**

- [x] All tests passed
- [ ] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: app/com/microsoft/azure/iotsolutions/iothubmanager/services/DeviceProperties.java [hunk 1]
```diff
@@ -21,40 +21,56 @@
 import java.util.concurrent.*;
 import java.util.stream.Collectors;
 
+/**
+ * @summary This class creates/reads cache of deviceProperties in/from CosmosDB.
+ * @remarks This is done to avoid request throttling when deviceProperties are queried directly from IOT-Hub.
```
Comment (sushilraje) [inline app/com/microsoft/azure/iotsolutions/iothubmanager/services/DeviceProperties.java]: The word is cache is confusing. generally, cache refers to in-memory cache OR external caches like Redis

File: app/com/microsoft/azure/iotsolutions/iothubmanager/services/DeviceProperties.java [hunk 2]
```diff
@@ -21,40 +21,56 @@
 import java.util.concurrent.*;
 import java.util.stream.Collectors;
 
+/**
+ * @summary This class creates/reads cache of deviceProperties in/from CosmosDB.
```
Comment (sushilraje) [inline app/com/microsoft/azure/iotsolutions/iothubmanager/services/DeviceProperties.java]: The word is cache is confusing. generally, cache refers to in-memory cache OR external caches like Redis

File: app/com/microsoft/azure/iotsolutions/iothubmanager/services/DeviceProperties.java [hunk 3]
```diff
@@ -96,12 +119,18 @@ public DeviceProperties(IStorageAdapterClient storageClient,
                     Json.fromJson(Json.parse(m.getData()), DevicePropertyServiceModel.class)
                 );
             } catch (ConflictingResourceException e) {
-                log.info("SetCacheAsync Conflicted ");
+                log.info("UpdateListAsync Conflicted ");
```
Comment (sushilraje) [inline app/com/microsoft/azure/iotsolutions/iothubmanager/services/DeviceProperties.java]: Shouldn't we be more explicit about "What" conflicted? I m not sure if stack trace OR logs would provide enough context to what went wrong here.  Our team members may get the What is the issue, but I doubt others would.

File: app/com/microsoft/azure/iotsolutions/iothubmanager/services/DeviceProperties.java [hunk 4]
```diff
@@ -152,65 +183,95 @@ public CompletionStage TryRecreateListAsync(boolean force) throws Exception {
...
     }
 
+    /**
+     * @return DeviceTwinName asynchronously
+     *
+     * @throws ExternalDependencyException
+     * @summary Get list of DeviceTwinNames from IOT-hub and whitelist it.
+     * @remarks List of Twin Names to be whitelisted is hardcoded in application.conf
+     */
     private CompletionStage<DeviceTwinName> getValidNamesAsync() throws ExternalDependencyException {
```
Comment (ppathan) [inline app/com/microsoft/azure/iotsolutions/iothubmanager/services/DeviceProperties.java]: Should the method names be camelCase or PascalCase?

File: app/com/microsoft/azure/iotsolutions/iothubmanager/services/DeviceProperties.java [hunk 5]
```diff
@@ -152,65 +183,95 @@ public CompletionStage TryRecreateListAsync(boolean force) throws Exception {
...
+    /**
+     * @param whitelist         Comma seperated list of deviceTwinName to be whitlisted which is hardcoded in
+     *                          appsettings.ini.
+     * @param fullNameWhitelist An out paramenter which is a list of deviceTwinName to be whitlisted without regex.
+     * @param prefixWhitelist   An out paramenter which is a list of deviceTwinName to be whitlisted with regex.
+     *
+     * @summary Parse the comma seperated string "whitelist" and create two separate list One with regex(*) and one
+     * without regex(*)
+     */
+    private void parseWhitelist(List<String> whitelist,
```
Comment (ppathan) [inline app/com/microsoft/azure/iotsolutions/iothubmanager/services/DeviceProperties.java]: All the method names in this file are mixed casing. Lets fix it.

File: app/com/microsoft/azure/iotsolutions/iothubmanager/services/DeviceProperties.java (hunk 2 above)
Comment (isaac73) [reply app/com/microsoft/azure/iotsolutions/iothubmanager/services/DeviceProperties.java]: Regardless of whether it is a DB cache /in-memory cache/redis cache, it is a cache defined by it's purpose. So, this should be OK.

File: app/com/microsoft/azure/iotsolutions/iothubmanager/services/DeviceProperties.java (hunk 4 above)
Comment (isaac73) [reply app/com/microsoft/azure/iotsolutions/iothubmanager/services/DeviceProperties.java]: done

File: app/com/microsoft/azure/iotsolutions/iothubmanager/services/DeviceProperties.java (hunk 5 above)
Comment (isaac73) [reply app/com/microsoft/azure/iotsolutions/iothubmanager/services/DeviceProperties.java]: done

File: app/com/microsoft/azure/iotsolutions/iothubmanager/services/DeviceProperties.java (hunk 3 above)
Comment (isaac73) [reply app/com/microsoft/azure/iotsolutions/iothubmanager/services/DeviceProperties.java]: done

**Pull request merged**